nilearn/experimental/surface/_io.py

```diff
@@ -37,3 +37,9 @@ def mesh_to_gifti(
     gifti_img.add_gifti_data_array(coords_array)
     gifti_img.add_gifti_data_array(faces_array)
     gifti_img.to_filename(gifti_file)
+
+
+def data_to_gifti(data: np.ndarray, gifti_file: pathlib.Path | str):
+    darray = gifti.GiftiDataArray(data=data, datatype="NIFTI_TYPE_FLOAT32")
```

Comment:
not sure if we want to always have NIFTI_TYPE_FLOAT32

Reply:
looked into it a bit and we have the following options
https://github.com/nipy/nibabel/blob/3b1c7b3727e18be0222b00dfe369bad6f3dc707d/nibabel/nifti1.py#L118

```python
_dtdefs = (  # code, label, dtype definition, niistring
    (0, 'none', np.void, ''),
    (1, 'binary', np.void, ''),
    (2, 'uint8', np.uint8, 'NIFTI_TYPE_UINT8'),
    (4, 'int16', np.int16, 'NIFTI_TYPE_INT16'),
    (8, 'int32', np.int32, 'NIFTI_TYPE_INT32'),
    (16, 'float32', np.float32, 'NIFTI_TYPE_FLOAT32'),
    (32, 'complex64', np.complex64, 'NIFTI_TYPE_COMPLEX64'),
    (64, 'float64', np.float64, 'NIFTI_TYPE_FLOAT64'),
    (128, 'RGB', np.dtype([('R', 'u1'), ('G', 'u1'), ('B', 'u1')]), 'NIFTI_TYPE_RGB24'),
    (255, 'all', np.void, ''),
    (256, 'int8', np.int8, 'NIFTI_TYPE_INT8'),
    (512, 'uint16', np.uint16, 'NIFTI_TYPE_UINT16'),
    (768, 'uint32', np.uint32, 'NIFTI_TYPE_UINT32'),
    (1024, 'int64', np.int64, 'NIFTI_TYPE_INT64'),
    (1280, 'uint64', np.uint64, 'NIFTI_TYPE_UINT64'),
    (1536, 'float128', _float128t, 'NIFTI_TYPE_FLOAT128'),
    (1792, 'complex128', np.complex128, 'NIFTI_TYPE_COMPLEX128'),
    (2048, 'complex256', _complex256t, 'NIFTI_TYPE_COMPLEX256'),
    (
        2304,
        'RGBA',
        np.dtype([('R', 'u1'), ('G', 'u1'), ('B', 'u1'), ('A', 'u1')]),
        'NIFTI_TYPE_RGBA32',
    ),
)
```

Reply:
oh nice!!!! thanks for this find
OK will use those

Reply:
at the moment I am skipping the intent part:

https://github.com/nipy/nibabel/blob/3b1c7b3727e18be0222b00dfe369bad6f3dc707d/nibabel/nifti1.py#L191

though seeing some it may be nice to maybe find a way to pass those when saving GLM output for example

Reply:
hmm actually only the following are supported for now:

https://github.com/nipy/nibabel/blob/3b1c7b3727e18be0222b00dfe369bad6f3dc707d/nibabel/gifti/gifti.py#L32-L36